typescript/infra/src/core/govern.ts

```diff
@@ -167,15 +169,21 @@ export class HyperlaneCoreGovernor<Chain extends ChainName> {
     const signer = connection.signer;
```

Comment:
This is outside this PR, but shouldn't this check be done sooner?

Reply:
I don't think so, care to elaborate?